src/proto/slp.rs

```diff
@@ -0,0 +1,197 @@
+//! MC Server List Ping protocol.
+//!
+//! Reference: http://wiki.vg/Server_List_Ping
+
+use std::fs::File;
+use std::io::ErrorKind::InvalidInput;
+use std::io::prelude::*;
+use std::io;
+use std::net::TcpStream;
+use std::ops::Sub; // Sub for Timespec
+use std::path::Path;
+
+use packet::{PacketRead, PacketWrite, Protocol};
+
+use rustc_serialize::base64::{ToBase64, STANDARD};
+use serde::json;
+use time;
+
+#[derive(Debug)]
+#[derive_serialize]
+#[derive_deserialize]
+pub struct Description {
+    // Actually this should be of type Chat
+    pub text: String,
+}
+
+#[derive(Debug)]
+#[derive_serialize]
+#[derive_deserialize]
+pub struct Players {
+    pub max: i32,
+    pub online: i32,
+    pub sample: Option<Vec<Sample>>,
+}
+
+#[derive(Debug)]
+#[derive_serialize]
+#[derive_deserialize]
+pub struct Sample {
+    pub name: String,
+    pub id: String,
+}
+
+#[derive(Debug)]
+#[derive_serialize]
+#[derive_deserialize]
+pub struct Version {
+    pub name: String,
+    pub protocol: i32,
+}
+
+/// Response sent to clients as JSON.
+#[derive(Debug)]
+#[derive_serialize]
+#[derive_deserialize]
+pub struct Response {
+    pub description: Description,
```

Comment:
Either here or in #37, whichever is merged later, we should fix this: the `Description` struct should be replaced by `ChatJson`.